Galactic's Monster Expansion

GalacticAC-Official opened the pull request:
Added 10 unique monsters to the game

sahlonahl committed:
Update README.md

PrincessPwny committed:
Update README.md

Prettify the URLs and add a server environment section

sahlonahl committed:
Merge pull request #1 from PrincessPwny/main

Update README.md

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

GalacticAC-Official committed:
Add files via upload

sahlonahl commented on Datapack/data/dvz/advancements/piglin_brute_shot.json:
```diff
@@ -0,0 +1,22 @@
+              "type": "minecraft:player"
+            },
+            "direct_entity": {
+              "type": "#minecraft:arrows"
```
I didn't even know you could do this!!?

sahlonahl commented on Datapack/data/dvz/functions/death_message.mcfunction:
```diff
@@ -43,6 +53,20 @@ execute as @a[scores={DVZ.death=1..},tag=johnny] run tellraw @a ["",{"text":"Joh
+execute as @a[scores={DVZ.death=1..},tag=endermite] run summon endermite ~ ~ ~ {Team:"zMONSTER"}
+execute as @a[scores={DVZ.death=1..},tag=endermite] run summon endermite ~ ~ ~ {Team:"zMONSTER"}
+
+##Spawn Magma Cube
```
This section might be a bit OP?

sahlonahl commented on Datapack/data/dvz/functions/rightclick/endermite.mcfunction:
```diff
@@ -0,0 +1,31 @@
+# -------------------------------------------
+# Called By: dvz:rightclick\detect
+# File Name: piglin brute
```
this is not the file name

sahlonahl commented on Datapack/data/dvz/functions/rightclick/evoker_old.mcfunction:
```diff
@@ -0,0 +1,34 @@
+# -------------------------------------------
+# Called By: dvz:rightclick
+# File Name: evoker
```
This file can be deleted.  Old evoker is being removed.

sahlonahl commented on Datapack/data/dvz/functions/tick.mcfunction:
```diff
@@ -350,6 +350,25 @@ execute as @a[tag=dwarves,scores={DVZ.icy.check=1..}] at @s run function dvz:zom
+execute as @a[tag=evoker,nbt={HurtTime:9s}] at @s run execute at @a[tag=dwarves,distance=..7.7] run summon evoker_fangs ~ ~ ~
+
+#Allay Loop
+execute as @a[tag=allay_carried] at @s run function dvz:zombies/allay/allay_loop
```
this may cause a lot of server lag with the amount of TPs

GalacticAC-Official replied:
Keeping it for now, will change if better method is found

sahlonahl commented on Datapack/data/dvz/functions/zombies/allay/allay_fly.mcfunction:
```diff
@@ -0,0 +1,24 @@
+# Called By: dvz:rightclick/detect
+# File Name: allay_fly
+# File Purpose: Moves allay until maximum distance or hits a wall
+# Created By: Zaffre
```
Not created by Zaffre

sahlonahl commented on Datapack/data/dvz/functions/zombies/enchanter/enchant_loop.mcfunction:
```diff
@@ -0,0 +1,25 @@
+execute as @s[scores={DVZ.enchanterloop.ticks=1..}] at @s run effect give @s glowing 1 0
+execute as @s[scores={DVZ.enchanterloop.ticks=1..}] at @s run effect give @s haste 1 2
+execute as @s[scores={DVZ.enchanterloop.ticks=1..}] at @s run effect give @s resistance 1 1
+execute as @s[scores={DVZ.enchanterloop.ticks=1..}] at @s run attribute @s minecraft:generic.attack_damage modifier add 23df59fa-29af-4c00-91dd-dad35afbb73d enchanter_damage 5 add
```
this is run every tick for 25 ticks. Perhaps a better way, so it's only applied once?

sahlonahl replied:
You could have it set at tick 25, with `enchant_pulse` and this file is only used to remove?

GalacticAC-Official replied:
Seems safer to put it on a loop, might cause unwanted behaviour otherwise

sahlonahl commented on Datapack/data/dvz/functions/zombies/enchanter/enchant_pulse.mcfunction:
```diff
@@ -0,0 +1,29 @@
+execute as @s if entity @s[tag=zenchanter,scores={DVZ.enchanterbook.cool=0}] run damage @s 1 minecraft:out_of_world
+
+execute as @s at @s if entity @s[tag=zenchanter,scores={DVZ.enchanterbook.cool=0}] run execute as @a[team=zMONSTER,distance=0.1..9,limit=1,sort=nearest,tag=!golem,tag=!enderman,tag=!miniboss,tag=!chargecrp,tag=!pig3,tag=!slime,tag=!evoker] at @s run scoreboard players set @s DVZ.enchanterloop.ticks 25
+execute as @s if entity @s[tag=zenchanter,scores={DVZ.enchanterbook.cool=0}] run scoreboard players set @s DVZ.enchanterbook.cool 1
```
1 second, huh?  Seems kinda low.

GalacticAC-Official replied:
Intended to be spammed as the buff effect lasts for a simillar period of time

sahlonahl commented on Datapack/data/dvz/functions/zombies/evoker/evoker_bolt_hit.mcfunction:
```diff
@@ -0,0 +1,22 @@
+particle minecraft:dust 0 0 0 2.5 ~ ~ ~ 1 1 1 0 75
+particle minecraft:end_rod ~ ~ ~ 1 1 1 0 75
+execute as @a[tag=dwarves,distance=..3.5] at @s run xp add @s -25 levels
+execute as @a[tag=dwarves,distance=..3.5] at @s run scoreboard players set @s evoker_curse 140
```
again, this can be set once so it's not applying effects and attributes every tick.  Only removes at the right time.

sahlonahl commented on Datapack/data/dvz/functions/zombies/evoker/evoker_summon.mcfunction:
```diff
@@ -0,0 +1,30 @@
+execute as @s if entity @s[tag=evoker,scores={DVZ.evokersummon.cool=0}] at @s run summon pillager ~ ~ ~ {Team:"zMONSTER",Tags:["legion"],CustomName:'{"text":"Illager Legionary"}',HandItems:[{id:"minecraft:crossbow",Count:1b,tag:{Enchantments:[{id:"minecraft:unbreaking",lvl:1s},{id:"minecraft:multishot",lvl:1s},{id:"minecraft:quick_charge",lvl:1s}]}},{id:"minecraft:tipped_arrow",Count:64b,tag:{Cus ...
+execute as @s if entity @s[tag=evoker,scores={DVZ.evokersummon.cool=0}] at @s run summon pillager ~ ~ ~ {Team:"zMONSTER",Tags:["legion"],CustomName:'{"text":"Illager Legionary"}',HandItems:[{id:"minecraft:crossbow",Count:1b,tag:{Enchantments:[{id:"minecraft:unbreaking",lvl:1s},{id:"minecraft:multishot",lvl:1s},{id:"minecraft:quick_charge",lvl:1s}]}},{id:"minecraft:tipped_arrow",Count:64b,tag:{Cus ...
+
+execute as @s if entity @s[tag=evoker,scores={DVZ.evokersummon.cool=0}] run scoreboard players set @s DVZ.evokersummon.cool 25
```
An item this powerful should probably have only 1 use.

GalacticAC-Official replied:
Sort of core identity of the class, increasing cooldown on all abilities for a nerf but will see if summons needs a nerf depending on how they work in combat

sahlonahl commented on Datapack/data/dvz/functions/zombies/piglin_brute/porkchop.mcfunction:
```diff
@@ -0,0 +1,18 @@
+# -------------------------------------------
+# Called By: dvz:zombies/piglin_brute/porkchop.mcfunction
```
set this to be called by the advancement

sahlonahl commented on Datapack/data/dvz/functions/zombies/piglin_brute/piglin_brute_shot_team.mcfunction:
```diff
@@ -0,0 +1,17 @@
+# -------------------------------------------
+# Called By: dvz:zombies/piglin_brute/piglin_brute_shot_team.mcfunction
```
set this to be called by the advancement...

sahlonahl commented on Datapack/data/dvz/functions/zombies/piglin_brute/piglin_brute_shot.mcfunction:
```diff
@@ -0,0 +1,20 @@
+# -------------------------------------------
+# Called By: dvz:zombies/piglin_brute/piglin_brute_shot.mcfunction
```
needs to be called by advancement

sahlonahl commented on Datapack/data/dvz/functions/zombies/slime/slime_bomb.mcfunction:
```diff
@@ -0,0 +1,22 @@
+# -------------------------------------------
+# Called By: detect
```
needs to be rightclick/detect

sahlonahl commented on Datapack/data/dvz/functions/zombies/slime/slime_launch.mcfunction:
```diff
@@ -0,0 +1,31 @@
+# Comments: 
+# -------------------------------------------
+
+summon area_effect_cloud ^ ^ ^1 {Tags:["slime_direction"]}
```
use markers wherever possible, not AOE clouds

sahlonahl reviewed (changes requested):
See individual comments.  Biggest issues are:
Lag options,
Naming conventions (specifically `called by` options),
and balance issues.

Please look at and consider all of these!
(I didn't have time to look through absolutely everything btw)

GalacticAC-Official committed:
Add files via upload

Fixed Allay Pickup feedback

GalacticAC-Official committed:
Add files via upload

Fixed allay pickup cooldown
updated some mob kits

GalacticAC-Official committed:
Add files via upload

Updated tick.mcfunction

GalacticAC-Official committed:
Loot table tweak

Increase weight of magma cube

GalacticAC-Official committed:
Header Fixes and More

Fixes header
Changed some mob abilities

GalacticAC-Official committed:
Update note

Updated update note

GalacticAC-Official committed:
Delete evoker_old.mcfunction

sahlonahl committed:
Modified Enchanter to be less laggy.
Changed "dvz:tick" to include more headers.

GalacticAC-Official committed:
Evoker Optimization

Changed evoker skill to be less laggy

GalacticAC-Official committed:
Evoker
- Warding does not work on self now
- Armor: Prot 4 -> Prot 3
Witch
- No longer have a sword
Magma Cube
- No longer have a sword
Enchanter
+ Ranged Increased
Piglin
- No longer have a sword
Vex
+ Armor: Feather Falling 2 -> Feather falling 10
Allay
> Now carries player above it instead of below

GalacticAC-Official committed:
Updated enchant_pulse.mcfunction

GalacticAC-Official committed:
Update slime_spit_hit.mcfunction

Fixed relative coords

GalacticAC-Official committed:
Update slime.mcfunction

GalacticAC-Official committed:
Update evoker.mcfunction

GalacticAC-Official committed:
Update evoker_summon.mcfunction

Nerfed evoker summon

GalacticAC-Official committed:
Update allay_pick.mcfunction

GalacticAC-Official committed:
Update allay_pick.mcfunction

GalacticAC-Official committed:
Create endermite_explode.mcfunction

Will fill in later

GalacticAC-Official committed:
Update tick.mcfunction

Updated allay fly for fail-proof and sneak disengage

GalacticAC-Official committed:
Update zenchanter.mcfunction

Changed name to mooshroom

GalacticAC-Official committed:
Update enchant_pulse.mcfunction

GalacticAC-Official committed:
Update mob_death.mcfunction

GalacticAC-Official committed:
Update mob_hurt.mcfunction

GalacticAC-Official committed:
Update assassinzombies.json

GalacticAC-Official committed:
Update dragonzombies.json

GalacticAC-Official committed:
Update guardianzombies.json

GalacticAC-Official committed:
Update witherzombies.json

GalacticAC-Official committed:
Update _enchanter_book.json

GalacticAC-Official committed:
Update piglin.mcfunction

GalacticAC-Official committed:
Update slime_spit_hit.mcfunction

GalacticAC-Official committed:
Update magma_cube.mcfunction

GalacticAC-Official committed:
Update magma_throw.mcfunction

GalacticAC-Official committed:
Update tick.mcfunction

GalacticAC-Official committed:
Update endermite.mcfunction

GalacticAC-Official committed:
Update tick.mcfunction

GalacticAC-Official committed:
Update endermite_explode.mcfunction

GalacticAC-Official committed:
Update endermite.mcfunction

GalacticAC-Official committed:
Update tick.mcfunction

GalacticAC-Official committed:
Update ai_buff.mcfunction

GalacticAC-Official committed:
Shulker and Bat Update
Added Shulker
Added Bat

GalacticAC-Official committed:
Update shulker_explodebomb.mcfunction

GalacticAC-Official committed:
Update shulker_explodetrap.mcfunction

GalacticAC-Official committed:
Update shulker_explodewall.mcfunction

GalacticAC-Official committed:
Bug Fixes
Fixed some shulker and bat related issues

GalacticAC-Official committed:
Nerfed Shulker Wall lifetime

GalacticAC-Official committed:
Further nerfed shulker wall lifetime

GalacticAC-Official committed:
Further nerfed shulker wall lifetime

GalacticAC-Official committed:
Update shulker_trapexplode.mcfunction

GalacticAC-Official committed:
Update shulker_wallexplode.mcfunction

GalacticAC-Official committed:
Update bat.mcfunction

GalacticAC-Official committed:
Balance Changes
Changed witch potion system
Minor Nerf to evoker
Should fix allay carrying golems

GalacticAC-Official committed:
Update allay_loop.mcfunction

GalacticAC-Official committed:
Update allay_pick.mcfunction

GalacticAC-Official committed:
Update allay_loop.mcfunction

GalacticAC-Official committed:
Merge branch 'sahlonahl:main' into OGvZ_GalacticAC

GalacticAC-Official committed:
Update tick.mcfunction

GalacticAC-Official committed:
Merge branch 'OGvZ_GalacticAC' into active-development

GalacticAC-Official committed:
Merge pull request #3 from GalacticAC-Official/active-development

Active development

GalacticAC-Official committed:
7/15 Balance Patch
Allay:
-Can now carry any mob
-You can either carry any number of normal mobs, or one important mob (golem, enderman etc)
Endermite
-Mine Count 3 -> 5
-Mine Explosion 2 -> 3
-Now have particles only visable to monster to prevent confusion
Evoker
-Now have permanent slowness 2
Mooshroom
-Cooldown 1s->2s
-Buff duration 2s->3s
-Can now buff any mobs
Piglin Brute
-Protection on each piece 2 -> 0
-Projectile Proection on each piece 1 -> 4
Vex
-Phantom Killer Damage Multiplier: 2x -> 3x
-Etheral Form Cooldown 20s -> 15s

GalacticAC-Official committed:
Fixed Endermite Particle not working

GalacticAC-Official committed:
Update tick.mcfunction

GalacticAC-Official committed:
Fixed Witch Potion

GalacticAC-Official committed:
Fixed Evoker Slowness

GalacticAC-Official committed:
Merge branch 'OGvZ_GalacticAC' into active-development

GalacticAC-Official committed:
Merge pull request #4 from GalacticAC-Official/active-development

Active development

GalacticAC-Official committed:
Patch on 2023/7/23
Endermite
- Endermine have reduced explosion power near shrine
Evoker
- Warding no longer give absorption to ai mobs
Piglin
- No longer have a flint & steel
Magma Cube
- Added a limited lifespan to all summons
Witch
- Witch buff potion instant healing 3->2
- Frog potion jump boost potency 3->2
- Frog potion duration 10min->1min

GalacticAC-Official committed:
Merge pull request #5 from GalacticAC-Official/temp

Patch on 2023/7/23

GalacticAC-Official committed:
Update detect.mcfunction

Undeleted bat_dark

GalacticAC-Official committed:
Update endermite_explode.mcfunction

GalacticAC-Official committed:
For Patch on 2023/8/6
Bat
- Echolocation have a smaller area but now deals minor damage
Endermite (Overhaul)
- Endermite now has a pretty overpowered sword, but it only have 5 durability
- However, it can be repaired on an anvil with a stole sword from dwarves
- The mines are reworked into providing different debuff and harassment effects (you also get more of them!)
- Weight lowered slightly
Evoker
- No longer drains mana
- Cursed bolt now deals minor damage
Magma Cube
- Lifespan halved on all summons
Piglin (Overhaul)
- All building material removed
- Weight lowered
- Gains explosive charges that can be placed and detonated
- Do plan carefully as the charges take time to prime!
Piglin Brute
- Total Armor 16->18
Shulker
- Wall conjured is slightly taller
- Bomb conjure cooldown 25s->15s
- Trap conjure removed and replaced with shulker bullet barrage (15s cooldown)
Vex
- Max health 14->20
- Sword Sharpmess 3->5
- Additionally has Projectile Protection 4 on helmet

GalacticAC-Official committed:
Update piglin.mcfunction

Fix error in item name

GalacticAC-Official committed:
Update shulker_throwtrap.mcfunction

Fixed Shulker Bullets

GalacticAC-Official committed:
Misc. Bugfixes

GalacticAC-Official committed:
Update evoker_summon.mcfunction

Removed TP legion on recast

GalacticAC-Official committed:
For Patch on 2023/08/13
Bat:
- Shroud of Night have slightly longer effect duration, and cleanses strength, speed and jump boost from dwarves, cooldown increased as compensation.
Endermite:
- Now starts with xp levels to repair its sword
- Endermine appearance slightly tweaked and health slightly lowered
- Displacement and Debuff mines are slightly weaker near shrine
Mooshroom:
- Buff mobs no longer glow
- The buff skill now have an effect area like a cone and all player monsters within the cone are buffed
Piglin:
- Explosive Charge has slightly lower health
Slime:
- Slime spit no longer damages blocks but deals higher damage

GalacticAC-Official committed:
Update endermite_explode.mcfunction

Fixed bug concerning displace/debuff mines

GalacticAC-Official committed:
Update endermite_explode.mcfunction

GalacticAC-Official committed:
Update endermite_explode.mcfunction

GalacticAC-Official committed:
Update endermite_explode.mcfunction
